src/acl/Data.h

```diff
@@ -21,12 +23,9 @@ class ACLData
 
     virtual ~ACLData() {}
 
-    /// \returns the flags supported by these ACL parameters (e.g., "-i")
-    virtual const Acl::ParameterFlags &supportedFlags() const { return Acl::NoFlags(); }
-
     virtual bool match(M) =0;
     virtual SBufList dump() const =0;
-    virtual void parse() =0;
+    virtual void parse(const ACL *) =0;
```

Comment:
Avoid using pointers for must-exist objects.

It feels a bit odd that the ACL is constant when we are parsing (i.e. modifying) its component.

```suggestion
    using Parse = DataParser;
    virtual void parse(Parser &) =0;
```

Since we have to modify all the ACLData::parse() methods, let's try to avoid similar future modifications by using parser parameter that we can enhance without adjusting all the methods. The new Acl::DataParser class can contain just two fields for now: A reference to the current parser and a reference to the current ACL object. The legacy parsing methods do not have to use the new parameter (or some of its fields) if they do not need them, of course.